hazelcast/src/main/java/com/hazelcast/collection/impl/queue/ForwardingQueueItemComparator.java

```diff
@@ -0,0 +1,53 @@
+package com.hazelcast.collection.impl.queue;
+
+import com.hazelcast.internal.serialization.SerializationService;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+/**
+ * A comparator which forwards its {@link Comparator#compare(Object, Object)} to another supplied comparator.
+ * If {@code comparator} is {@code null}, then {@link Comparator#compare(Object, Object)} uses {@link QueueItem#compareTo(QueueItem)}
+ *
+ * @param <T> the type of objects that may be compared by the delegated comparator as defined by the supplied comparator instance in {QueueConfig}
+ */
+public final class ForwardingQueueItemComparator<T> implements Comparator<QueueItem>, Serializable {

```

Comment:
Why is it `Serializable`? If it needs to be sent between nodes, we'll need to ensure that the serialization service is set after deserializing instead of it being `null` and I suggest we add support for some of hazelcasts' better serialization strategies and not java serialization.

Reply:
Thanks to point that out. I don't see any need for serialization and will remove this.